### PR TITLE
Building druid-it-tools and running for travis in it.sh

### PR DESCRIPTION
**Pull request by abhagraw**
Building druid-it-tools before the new ITs are run

### CODE REVIEW

**Comment by paul-rogers** [comment]
@abhagraw, thanks for the fix. Let's also fix the issue that allowed the task to pass though the code itself failed. I suspect we need to add `set -e` to the script (cause the script to fail if a command fails.)

**Comment by paul-rogers** [comment]
The fix is good, but incurs the cost of building the IT tools every time. (Bad enough we have to rebuild the image.) A better fix is in `.travis.yml`, line 49:

```text
install...' && MAVEN_OPTS='-Xmx3000m' travis_wait 15 ${MVN} clean install -q -ff -pl '!distribution,!:druid-it-tools,!:druid-it-image,!:druid-it-cases' ${MAVEN_SKIP} ${MAVEN_SKIP_TESTS} -T1C && ${MVN} install -q -ff -pl 'distribution' ${MAVEN_SKIP} ${MAVEN_SKIP_TESTS}
```

Remove the `druid-it-tools` exclusion:

```text
install...' && MAVEN_OPTS='-Xmx3000m' travis_wait 15 ${MVN} clean install -q -ff -pl '!distribution,!:druid-it-image,!:druid-it-cases' ${MAVEN_SKIP} ${MAVEN_SKIP_TESTS} -T1C && ${MVN} install -q -ff -pl 'distribution' ${MAVEN_SKIP} ${MAVEN_SKIP_TESTS}
```

Also, it seems that the above does build a distribution, so we can skip that step in `it.sh`. So, `it.sh` would change to:

```text
    "travis" )
        ...
    	$0 image
    	$0 test $1
    	$0 tail $1
```

That is, remove `dist` (seems to already be done) and `tools` (since that will now be done in the Maven build.)

**Comment by paul-rogers** [comment]
Very nice! Thanks for catching the other issues, such as the old `it-tools` names.
LGTM (non-binding)

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/config/ClusterConfigTest.java [hunk 1]
```diff
@@ -85,14 +85,16 @@ public void testYaml() throws FileNotFoundException
...
-    File userEnv = new File(
-        new File(
-            System.getProperty("user.home"),
-            "druid-it"),
-        "Test.env");
-    try (PrintWriter out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(userEnv), StandardCharsets.UTF_8))) {
-      out.println("druid_user_var=user");
-    }
+    // This test is user-specific. Enable it only if this particular feature changes.
+    //    File userEnv = new File(
```
Comment (cryptoe) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/config/ClusterConfigTest.java:89]: Shouldn't this be removed all together ?

**Review by abhishekagarwal87 — APPROVED** [review]

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/config/ClusterConfigTest.java [hunk 2]
```diff
@@ -85,15 +79,6 @@ public void testYaml() throws FileNotFoundException
     assertEquals("http://localhost:8888", service.clientUrl());
     assertEquals("http://localhost:8888", resolved.routerUrl());
```
Comment (paul-rogers) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/config/ClusterConfigTest.java:81]: Can I ask you to comment this out and not remove it? The functionality needs to be tested; let's not use the code itself.

File: it.sh [hunk 3]
```diff
@@ -18,17 +18,20 @@
...
 # Utility script for running the new integration tests, since the Maven
 # commands are unwieldy.
+set -e
 
 export DRUID_DEV=$(cd $(dirname $0) && pwd)
 
 function usage
 {
-	cat <<EOF
+  cat <<EOF
```
Comment (paul-rogers) [inline it.sh:27]: What is our standard indentation in scripts? It is 2 in Java. I think I saw it as 4 in other scripts, though I might be mistaken.

**Review by paul-rogers — COMMENTED** [review]
A couple of minor comments, else LGTM. (Non-binding).

File: it.sh (hunk 3 above)
Comment (abhagraw) [reply it.sh:27]: There was a mix of tabs and spaces in the script. I just replaced all the tabs with spaces to keep it consist. I think, IntelliJ decided it should to 2 spaces for shell script.

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/config/ClusterConfigTest.java (hunk 2 above)
Comment (abhagraw) [reply integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/config/ClusterConfigTest.java:81]: Will make this change in the next PR, which will a fix for the batch index test which were disabled here.
Would not like to disturb the green build for this change.

**Pull request merged**